tests/support/unit.py

```diff
@@ -157,7 +157,10 @@ def run(self, result=None):
                             'Please consider using \'del self.%s\' on the test case '
                             '\'tearDown()\' method', self.__class__.__name__, attr,
                             getattr(self, attr), attr)
-                delattr(self, attr)
+                if hasattr(self.__class__, attr):
```

Comment:
Why was this needed?
If it started failing then it's something related to this PR or we'd have way PRs failing because of it.

A link to one of the failed builds would help understand the reason why.

Reply:
Yeah, this is actually needed because `dir` lists also class attributes as well (and `delattr` fails on those). I initially had the cleanup in tests (because on Win there was new `__slots__` attribute), but I thought it's better to fix it for any other usecases.